src/evict/evict_lru.c

```diff
@@ -1029,21 +1041,10 @@ retry:	while (slot < max_entries && ret == 0) {
 		    conn->hazard_max - WT_MIN(conn->hazard_max / 2, 10))
 			continue;
 
-		/*
-		 * If we are filling the queue, skip files that haven't been
-		 * useful in the past.
-		 */
-		if (btree->evict_walk_period != 0 &&
-		    cache->evict_entries >= WT_EVICT_WALK_INCR &&
-		    btree->evict_walk_skips++ < btree->evict_walk_period)
-			continue;
-		btree->evict_walk_skips = 0;
 		prev_slot = slot;
 
 		(void)WT_ATOMIC_ADD4(dhandle->session_inuse, 1);
 		incr = 1;
-		__wt_spin_unlock(session, &conn->dhandle_lock);
-		dhandle_locked = 0;
```

Comment:
This change isn't feasible: we do a lot of work to avoid holding `dhandle_lock` when not actively accessing the list of data handles.  I think it should be dropped here and acquired again if the handle needs to be moved to a different place in the list.